### ISSUE TITLE
Add space character to Typewriter font

### ISSUE THREAD

**Issue by k4b7** [Khan/MathJax-dev#4]
After looking at `makeFF` for a bit, it appears that no fonts have the standard `0x20` space character.  There are more specific space characters defined in Main-Regular and Main-Bold, e.g. 
```
   spaceEn => [
    'Select(0u2002)',
    'SetRBearing(500)',
  ],

  spaceEm => [
    'Select(0u2003)',
    'SetRBearing(999)',
  ],
...
```
I verified that these spaces actually exist in the font by replace a `1` on the KaTeX demo page with a `\u2002` and then `\u2003` using the console and say the `1` turn to a space and then double in width.

I think we should be able to a space to Typewriter by creating an `$extra` entry in `makeFF` like so:
```
$extra{'Typewriter'} = {
   space => [
     'Select(0u20)',
     'SetRBearing(525)',  // advance for all of the characters in Typewriter
   ],
}
```

**Comment by gagern** [comment]
Sounds like a good approach. Want to write a PR for this? Do we want to add spaces to other fonts, too, in order to ensure consistent layout no matter what fallback font a system may have?

**Comment by k4b7** [comment]
I can write a PR for this.  I don't understand the concern about fallback fonts... if the browser fails to load Typewriter is going to use a system font not one of the other KaTeX fonts.

**Comment by gagern** [comment]
No, the concern was not for Typewriter, but for regular `\text{foo bar}`. Or were you only referring to the tpyewriter width version when you wrote that “no fonts have the standard `0x20` space character”?

**Comment by k4b7** [comment]
I see.  For some reason I thought we were using CSS to creating spaces in text mode.  Yes we should definitely add space characters to each of the fonts that's used by one of the text mode commands.
